processing/src/main/java/io/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -60,14 +60,30 @@
    *
    * @return a new Factory that can be used for operations on top of data output from the current factory.
    */
-  public AggregatorFactory getCombiningFactory();
+  public abstract AggregatorFactory getCombiningFactory();
+
+  /**
+   * Returns an AggregatorFactory that can be used to merge the output of aggregators from this factory and
+   * other factory.
+   * This method is relevant only for AggregatorFactory which can be used at ingestion time.
+   *
+   * @return a new Factory that can be used for merging the output of aggregators from this factory and other.
+   */
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
```

Comment:
what is the expected behavior at merge time if the aggregator factory throws AggregatorFactoryNotMergeableException? is there a fallback, or does it fail?